tests/test_fairworkflow.py

```diff
@@ -287,25 +305,64 @@ def test_draw_with_graphviz_module_without_dependency(self, tmp_path):
 
         with mock.patch.dict('sys.modules', {'graphviz': mock_graphviz}):
             with pytest.raises(RuntimeError):
-                self.workflow.draw(filepath=str(tmp_path))
+                test_workflow.draw(filepath=str(tmp_path))
 
-    def test_draw_with_graphviz_module_and_dependency(self, tmp_path):
+    def test_draw_with_graphviz_module_and_dependency(self, tmp_path, test_workflow):
         """
         Test draw method with graphviz python module and graphviz software
         installed
         """
-        self.workflow.draw(filepath=str(tmp_path))
+        test_workflow.draw(filepath=str(tmp_path))
 
     @mock.patch.dict('sys.modules', {'graphviz': None})
-    def test_display_without_graphviz_module(self):
+    def test_display_without_graphviz_module(self, test_workflow):
         """Test display method without graphviz python module installed."""
         with pytest.raises(ImportError):
-            self.workflow.display()
+            test_workflow.display()
 
-    def test_display_with_graphviz_module_and_dependency(self):
+    def test_display_with_graphviz_module_and_dependency(self, test_workflow):
         """
         Test display method with graphviz python module and graphviz software
         installed
         """
-        self.workflow.display()
+        test_workflow.display()
 
+    @mock.patch('fairworkflows.rdf_wrapper.NanopubClient.publish')
+    def test_publish_as_nanopub(self, mock_publish, test_workflow):
+        test_published_uris = ['www.example.org/published_step1#step',
+                               'www.example.org/published_step2#step',
+                               'www.example.org/published_step3#step',
+                               'www.example.org/published_workflow#workflow']
+        mock_publish.side_effect = [
+            {'concept_uri': test_published_uris[0]},  # first call
+            {'concept_uri': test_published_uris[1]},
+            {'concept_uri': test_published_uris[2]},
+            {'concept_uri': test_published_uris[3]}   # Last call
+        ]
+        for step in test_workflow:
+            assert step.is_modified
+        test_workflow.publish_as_nanopub()
```

Comment:
Do we need to check the returned publication_info too?

Reply:
No, because the publish method is mocked and publication_info is just what we tell it to return here:
https://github.com/fair-workflows/fairworkflows/blob/6e58c8a4911b90b4d591aa68530f07499757611b/tests/test_fairworkflow.py#L336

Reply:
Do we need to make this more clear?

Reply:
Ah no, sorry, I hadn't noticed it was mocked. I suppose my comment was triggered by seeing that the other new test (```test_publish_as_nanopub_no_modifications()```) does check the returned publication_info whereas this test doesn't. Both tests have a mocked client so it just looked asymmetric. Not really a big issue or anything.

Reply:
Well, I added the assertion anyway :) It is kind of trivial, but a bug could in theory be introduced there. And at least the confusion that you had wouldn't happen again!